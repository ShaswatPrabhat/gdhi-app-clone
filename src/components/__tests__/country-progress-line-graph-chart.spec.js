import { mount } from "@vue/test-utils";
import { describe, beforeEach, it, expect, vi } from "vitest";
import CountryProgressLineChart from "../graphs/country-progress-line-graph/country-progress-line-graph-chart.vue";
import Chart from "chart.js/auto";

describe("Country Progress Line Chart", () => {
  let wrapper;
  vi.mock("chart.js/auto");
  beforeEach(() => {
    wrapper = mount(CountryProgressLineChart, {
      propsData: {
        currentYear: "2023",
        defaultYear: "2022",
        categoryFilter: 0,
        locale: "en",
        countryName: "India",
        xAxisLabels: ["2020", "2021", "2022", "2023", "2024"],
        yearOnYearData: [
          {
            year: "2023",
            data: {
              country: {
                countryId: "IND",
                countryName: "India",
                countryAlpha2Code: "IN",
                categories: [],
                countryPhase: 3,
                updatedDate: "",
              },
              average: {},
            },
          },
        ],
      },
    });
  });
  it("should render Graph when mounted and create a new Chart instance", () => {
    expect(wrapper.vm.$el).toMatchSnapshot();
    expect(Chart.mock.calls.length).toEqual(1);
    expect(Chart.mock.calls[0][1]).toMatchSnapshot();
  });
});
