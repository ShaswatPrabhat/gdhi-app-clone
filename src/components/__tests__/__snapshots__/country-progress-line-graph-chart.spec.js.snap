// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Country Progress Line Chart > should render Graph when mounted and create a new Chart instance 1`] = `
<div
  class="line-chart-container"
  data-v-d2870135=""
  defaultyear="2022"
  selectedregion="[object Object]"
>
  <canvas
    data-v-d2870135=""
    id="line-chart"
  />
</div>
`;

exports[`Country Progress Line Chart > should render Graph when mounted and create a new Chart instance 2`] = `
{
  "data": {
    "datasets": [
      {
        "borderColor": "black",
        "borderWidth": 1.5,
        "data": [
          null,
          null,
          null,
          null,
          null,
        ],
        "fill": false,
        "label": "India",
        "pointBackgroundColor": [Function],
        "pointBorderColor": "transparent",
        "pointHoverRadius": 16,
        "pointRadius": 14,
        "pointStyle": "rectRot",
        "spanGaps": true,
      },
      {
        "borderColor": "#6C757D",
        "borderDash": [
          5,
          4,
        ],
        "borderWidth": 1.5,
        "data": [
          null,
          null,
          null,
          null,
          null,
        ],
        "fill": false,
        "label": "Global average",
        "pointBackgroundColor": "black",
        "pointHoverRadius": 8,
        "pointRadius": 1,
        "pointStyle": "circle",
        "spanGaps": true,
      },
    ],
    "labels": [
      "2020",
      "2021",
      "2022",
      "2023",
      "2024",
    ],
  },
  "options": {
    "animation": false,
    "clip": false,
    "layout": {
      "padding": {
        "bottom": 0,
        "left": 60,
        "right": 80,
        "top": 60,
      },
    },
    "maintainAspectRatio": false,
    "plugins": {
      "annotation": {
        "annotations": {
          "line": {
            "borderColor": "#415BA3",
            "borderDash": [
              20,
              16,
            ],
            "borderWidth": 1.5,
            "drawTime": "beforeDatasetsDraw",
            "type": "line",
            "xMax": "2023",
            "xMin": "2023",
          },
        },
      },
      "legend": {
        "labels": {
          "boxHeight": 0,
        },
        "position": "bottom",
        "rtl": false,
      },
      "tooltip": {
        "rtl": false,
      },
    },
    "responsive": true,
    "scales": {
      "x": {
        "border": {
          "color": "#ced4da",
          "dash": [
            2,
            4,
          ],
          "width": 1.5,
        },
        "grid": {
          "display": true,
          "drawTicks": true,
          "lineWidth": 0,
          "tickColor": "white",
          "tickLength": 16,
          "tickWidth": 1.5,
        },
        "reverse": false,
        "ticks": {
          "beginAtZero": true,
          "color": "black",
          "font": {
            "weight": "bold",
          },
          "min": 1,
          "stepSize": 1,
        },
      },
      "y": {
        "border": {
          "color": "#ced4da",
          "dash": [
            2,
            2,
          ],
          "width": 1,
        },
        "grid": {
          "drawTicks": true,
          "lineWidth": 1,
          "tickColor": "white",
          "tickLength": 16,
        },
        "max": 5,
        "min": 0,
        "position": "left",
        "ticks": {
          "beginAtZero": false,
          "callback": [Function],
          "color": "black",
          "stepSize": 1,
        },
      },
    },
  },
  "type": "line",
}
`;
